src/components/Modal/src/Modal.vue

```diff
@@ -7,6 +7,55 @@
 	</div>
 </template>
 
+<script>
+import modalApi from './modal-api';
+
+export default {
+	inject: {
+		modalApi,
+	},
+
+	props: {
+		/**
+		 * Close the modal on ESC
+		 */
+		closeOnEsc: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	mounted() {
+		this.addEvents();
+	},
+
+	destroyed() {
+		this.removeEvents();
+	},
+
+	methods: {
+		addEvents() {
+			if (this.closeOnEsc) {
+				document.addEventListener('keyup', this.handleEscKey);
+			}
+		},
+
+		removeEvents() {
+			if (this.closeOnEsc) {
+				document.removeEventListener('keyup', this.handleEscKey);
+			}
+		},
+
+		handleEscKey(event) {
+			const isClosingStackedModal = !!this.modalApi.state.vnode;
```

Comment:
I'm assuming since this is here you've tested that this prop works and behaves correctly with stacked nested modals, right?

Reply:
Yup

Reply:
Doesn't seem to be the case in the Stacked Modal demo: https://square.github.io/maker/styleguide/close-modal-on-esc/#/Modal